ping/ping_common.c

```diff
@@ -903,7 +903,12 @@ int finish(struct ping_rts *rts)
 #endif
 		printf(_(", %g%% packet loss"),
 		       (float)((((long long)(rts->ntransmitted - rts->nreceived)) * 100.0) / rts->ntransmitted));
-		printf(_(", time %ldms"), 1000 * tv.tv_sec + (tv.tv_nsec + 500000) / 1000000);
+
+        if (sizeof(time_t) == 4) { // 32 bits time_t
+		    printf(_(", time %lums"), (unsigned long)(1000 * tv.tv_sec + (tv.tv_nsec + 500000) / 1000000));
+        } else { // 64 bits time_t
+		    printf(_(", time %llums"), (unsigned long long)(1000 * tv.tv_sec + (tv.tv_nsec + 500000) / 1000000));
+        }
```

Comment:
I do not think that we need the condition. I think that we can just cast everything into unsigned long long and be done with it, i.e. we only need the:

```
printf(_(", time %llums"), (unsigned long long)(1000 * tv.tv_sec + (tv.tv_nsec + 500000) / 1000000));
```

Reply:
@Chozmao could you please test it? I should also add some 32bit build into CI.

Reply:
@pevik @metan-ucw , you are correct. I have tested the casting to unsigned long long on 32bit device with 32bit time_t. It also works as expected. I just updated my pull request. Thanks